src/constructs/core/stack.test.ts

```diff
@@ -63,4 +64,21 @@ describe("The GuStack construct", () => {
 
     expect(stack.app).toBe("MyApp");
   });
+
+  it("should return a parameter that exists", () => {
+    const stack = new GuStack(new App(), "Test", { app: "MyApp" });
+    const testParam = new GuParameter(stack, "MyTestParam", {});
+    stack.setParam(testParam);
+
+    const actual = stack.getParam<GuParameter>("MyTestParam");
```

Comment:
Probably a continuation of me misunderstanding the use case, but I'm seeing an opportunity to simplify the call to this getter by making it return GuParameter as a default, and leave the client to cast it if they need another type. Is getting a GuParameter sufficient for most cases?

```ts
const actual = stack.getParam("MyTestParam");
```

```ts
const actual = stack.getParam("MyTestParam") as GuSSMParameter;
```

Reply:
This syntax is still possible. The default type of `getParam` is `GuParameter`. You can do both of these:

```typescript
const foo = scope.getParam("Foo") as GuAmiParameter;
const bar = scope.getParam<GuAmiParameter>("Bar");
```

Both `foo` and `bar` will be of type `GuAmiParameter`. Will rework the tests to make this more obvious.

Reply:
Hmm, the linter isn't happy with the assertion variant (see [this rule](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-unnecessary-type-assertion.md)), so the generics version is the only way to cast. I think?